Maxar-Corp/UMS-flowable-spring-boot
# WIP: initial implementation of data seed for ums tasks

This initial implementation can serve as a talking point for a discussion:

Functionally, this loads some UmsTasks that have CRUD operations available. The data model needs iterated on to include all the fields listed in the Task/Request Form Page.

There is a placeholder for interacting with the workflow engine that is by passed for now until we decide how closely coupled the UMS api will be with the workflow. 

To Test:

1. Import the `UMS_API_Examples.postman_collection.json` to Postman or be prepped to replicate those on your own.
2. May need to reimport maven project from`pom.xml`
3. Start up/run the FlowableApplication.java
4. Check `http://localhost:8443/ums-api/h2-console` and log in with credentials defined in `application.properties`
5. Verify `UMS_TASK` created
6. Verify calls from imported Postman Collection execute as expected.

TO DO and/or discuss:

- [x] Use of Enums for limited data values
- [ ] Expand UmsTask data model to include all fields from task request form
- [x] Add programmatic tests per team convention - what level of tests are typical for: unit, integration, functional
- [ ] Add logging statements per team convention
